### PR TITLE
[JENKINS-44114] - Update to remoting 3.10 and add workdir support

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Upgrade Remoting from 3.7 to 3.10 ([Changelog](https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md)). It includes some bugfixes, improves logging options, and ads Work Directory support
- [x] - Reserve a space for Remoting Work Directory and expand it as a volume
- [x] - Document usage of Work directories in README

@reviewbybees

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,18 @@ This image may instead be used to launch an agent using the **Launch method** of
 docker run -i --rm --name agent jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar
 ```
 
-after setting **Remote root directory** to `/home/jenkins`.
+after setting **Remote root directory** to `/var/lib/jenkins/agent`.
```
Comment (jglick) [inline README.md]: I see `/home/jenkins/agent` above and below.

File: Dockerfile [hunk 2]
```diff
@@ -28,14 +28,17 @@ RUN groupadd -g 10000 jenkins
...
 
 RUN curl --create-dirs -sSLo /usr/share/jenkins/slave.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
   && chmod 755 /usr/share/jenkins \
   && chmod 644 /usr/share/jenkins/slave.jar
 
-
 USER jenkins
-RUN mkdir /home/jenkins/.jenkins
+ENV AGENT_WORKDIR=${AGENT_WORKDIR}
+RUN mkdir /home/jenkins/.jenkins && mkdir -p ${AGENT_WORKDIR}
```
Comment (jglick) [inline Dockerfile]: Do we still need `~/.jenkins`? What is it for?

File: README.md (hunk 1 above)
Comment (oleg-nenashev) [inline README.md]: My bad. I changed my mind about the directory name, but have not updated docs

File: Dockerfile (hunk 2 above)
Comment (oleg-nenashev) [inline Dockerfile]: For compatibility with use-cases, which use the agent without `-workDir` definition (and bravely use the latest image from DockerHub).

File: README.md [hunk 3]
```diff
@@ -19,7 +19,18 @@ This image may instead be used to launch an agent using the **Launch method** of
...
+
+### Agent Work Directories
+
+Starting from [Remoting 3.8](https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md#38) there is a support of Work directories, 
+which provides logging by default and change the JAR Caching behavior.
+
+Call example:
+
+```sh
+docker run -i --rm --name agent1 --v agent1-workdir:/home/jenkins/agent jenkinsci/slave:3.10-1 java -jar /usr/share/jenkins/slave.jar -workDir /home/jenkins/agent
```
Comment (jglick) [inline README.md]: You have to specify `-workDir` even though this is the default location defined in the image?

Comment (oleg-nenashev) [inline README.md]: Yes, due to the compatibility reasons

**Pull request merged**